crates/sdk/src/sys.rs

```diff
@@ -1,32 +1,51 @@
 #![allow(dead_code)]
 
 #[cfg(not(target_arch = "wasm32"))]
-mod mock;
 mod types;
 
 pub use types::*;
 
-#[link(wasm_import_module = "env")]
-extern "C" {
-    pub fn panic(loc: Location) -> !;
-    pub fn panic_utf8(msg: Buffer, loc: Location) -> !;
-    // --
-    pub fn register_len(register_id: RegisterId) -> PtrSizedInt;
-    pub fn read_register(register_id: RegisterId, buf: BufferMut) -> Bool;
-    // --
-    pub fn input(register_id: RegisterId);
-    pub fn value_return(value: ValueReturn);
-    pub fn log_utf8(msg: Buffer);
-    pub fn emit(event: Event);
-    // --
-    pub fn storage_read(key: Buffer, register_id: RegisterId) -> Bool;
-    pub fn storage_write(key: Buffer, value: Buffer, register_id: RegisterId) -> Bool;
+macro_rules! wasm_imports {
+    ($module:expr => { $(fn $func_name:ident($($arg:ident: $arg_ty:ty),*) $(-> $returns:ty)?;)* }) => {
+        cfg_if::cfg_if! {
+            if #[cfg(target_arch = "wasm32")] {
+                #[link(wasm_import_module = $module)]
+                extern "C" {
+                    $(
+                        pub fn $func_name($($arg: $arg_ty),*) $(-> $returns)?;
+                    )*
+                }
+            } else {
+                $(
+                    #[no_mangle]
+                    pub extern "C" fn $func_name($($arg: $arg_ty),*) $(-> $returns)? {
```

Comment:
since we're using this model, we no longer need the `extern "C"`, and to match the behaviour of the `extern`, we can mark this `unsafe`

```suggestion
                    pub unsafe fn $func_name($($arg: $arg_ty),*) $(-> $returns)? {
```